### PR TITLE
feat(BILL-CPC-1196): filter all bills by customer id for admin

### PR DESCRIPTION
**Pull request by PaoloOabel**
**JIRA:** link to jira ticket
https://champlainsaintlambert.atlassian.net/browse/CPC-1196

## Context:
Implemented a search bar to filter all the bills by customer ID. Also added customer ID to the table to facilitate searches.

## Does this PR change the .vscode folder in petclinic-frontend?:
No it does not.

<span style="color:red">**Reviewers need to check for any changes to the 
.vscode folder and add a comment about it to their review comments.**</span>

## Changes
- Implemented a filter in the admin's table of bills.
- Installed the fontawesome plugin Adem mentioned.

## Before and After UI (Required for UI-impacting PRs)
Before: 
beforeFilter

After:
image

### CODE REVIEW

**Review by Futurespast — DISMISSED** [review]
Good. The only thing is, I have a feeling we will have merge conflicts for the other PRs

**Review by ctsaki25 — DISMISSED** [review]
Looks good but can be improved. Next sprint we'll incorporate a single search bar. Also, are the new dependencies in package.json used/needed?

**Comment by PaoloOabel** [comment]
> Looks good but can be improved. Next sprint we'll incorporate a single search bar. Also, are the new dependencies in package.json used/needed?

Yeah in the next sprint let's focus on making the page cleaner and more uniform. The changes on the packages were undone.

File: petclinic-frontend/src/features/bills/AdminBillsListTable.tsx [hunk 1]
```diff
@@ -115,7 +124,7 @@ export default function AdminBillsListTable(): JSX.Element {
   };
 
   const handleDelete = async (billId: string): Promise<void> => {
-    const billToDelete = billsList.find(bill => bill.billId === billId);
+    const billToDelete = billsList.find((bill: Bill) => bill.billId === billId);
```
Comment (ctsaki25) [inline petclinic-frontend/src/features/bills/AdminBillsListTable.tsx:127]: glad this was caught. thanks!

File: petclinic-frontend/src/features/bills/AdminBillsListTable.tsx [hunk 2]
```diff
@@ -165,6 +174,18 @@ export default function AdminBillsListTable(): JSX.Element {
...
+  function isKeyOfBillResponseDTO(key: string): key is keyof Bill {
+    return ['customerId'].includes(key);
+  }
+
+  const filteredBills = billsList.filter((bill: Bill) => {
+    return Object.keys(filter).every(key => {
+      if (!filter[key]) return true;
+      if (!isKeyOfBillResponseDTO(key)) return true;
+      return bill[key].toString().includes(filter[key].toString());
+    });
```
Comment (ctsaki25) [inline petclinic-frontend/src/features/bills/AdminBillsListTable.tsx:186]: good implementation.

**Review by ctsaki25 — DISMISSED** [review]
Great work. Your implementation is good.

**Review by MercuryAtom31 — DISMISSED** [review]
Well done! Everything works.

**Comment by ctsaki25** [comment]
looks good, but I suggest you change "Filter by customerId" to "Search by Customer ID" as it makes more sense from the end users perspective (following User-Centric design)

**Comment by PaoloOabel** [comment]
> looks good, but I suggest you change "Filter by customerId" to "Search by Customer ID" as it makes more sense from the end users perspective (following User-Centric design)

Changed it! Thanks for ensuring the frontend is user focused.

**Review by ctsaki25 — APPROVED** [review]
Good work! Glad you followed User-Centric Design.

**Review by Futurespast — APPROVED** [review]

**Pull request merged**